src/vs/workbench/contrib/terminal/browser/terminalActions.ts

```diff
@@ -1462,44 +1459,24 @@ export function registerTerminalActions() {
 				terminalService.setActiveTabByIndex(Number(indexMatches[1]) - 1);
 				return terminalService.showPanel(true);
 			}
-			const detectedProfiles = await terminalService.getAvailableProfiles();
-
-			// Remove 'New ' from the selected item to get the profile name
-			const profileSelection = item.substring(4);
 			const customType = terminalContributionService.terminalTypes.find(t => t.title === item);
 			if (customType) {
 				return commandService.executeCommand(customType.command);
 			}
-			if (detectedProfiles) {
-				let launchConfig = detectedProfiles?.find((profile: { profileName: string; }) => profile.profileName === profileSelection);
-				if (launchConfig && !launchConfig.isWorkspaceProfile) {
-					const instance = terminalService.createTerminal({ executable: launchConfig.path, args: launchConfig.args });
-					terminalService.setActiveInstance(instance);
-				} else if (launchConfig && launchConfig.isWorkspaceProfile) {
-					if (!storageService.getBoolean(`${QUICK_LAUNCH_PROFILE_CHOICE}-${item}-${launchConfig.path}`, StorageScope.WORKSPACE, false)) {
-						notificationService.prompt(Severity.Info, `Do you allow this workspace to modify your terminal profile? ${item} with path ${launchConfig?.path}`,
-							[{
-								label: 'Allow',
-								run: () => {
-									const instance = terminalService.createTerminal({ executable: launchConfig?.path, args: launchConfig?.args });
-									terminalService.setActiveInstance(instance);
-									storageService.store(`${QUICK_LAUNCH_PROFILE_CHOICE}-${item}-${launchConfig?.path}`, true, StorageScope.WORKSPACE, StorageTarget.USER);
-								}
-							},
-							{
-								label: 'Disallow',
-								run: () => {
-									const activeInstance = terminalService.getActiveInstance();
-									if (activeInstance) {
-										terminalService.setActiveInstance(activeInstance);
-									}
-								}
-							}]
-						);
-					} else {
-						const instance = terminalService.createTerminal({ executable: launchConfig?.path, args: launchConfig?.args });
+
+			const quickSelectProfiles = await terminalService.getAvailableProfiles();
+
+			// Remove 'New ' from the selected item to get the profile name
+			const profileSelection = item.substring(4);
+			if (quickSelectProfiles) {
+				const launchConfig = quickSelectProfiles.find(profile => profile.profileName === profileSelection);
+				if (launchConfig) {
+					if (!terminalService.configHelper.checkWorkspaceShellPermissions(undefined, launchConfig.isWorkspaceProfile)) {
```

Comment:
I think `checkWorkspaceShellPermissions` should return true if it's ok to launch, it seems to be inverted?